includes/integrations/class.llms.integration.buddypress.php

```diff
@@ -54,79 +78,175 @@ protected function configure() {
 	}
 
 	/**
-	 * Add LLMS navigation items to the BuddyPress User Profile
+	 * Retrieve integration settings.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_integration_settings() {
+
+		$settings = array();
+
+		if ( $this->is_available() ) {
+
+			$endpoints = $this->get_profile_endpoints( false );
+
+			$display_eps = array();
+
+			foreach ( $endpoints as $ep_key => $endpoint ) {
+				$display_eps[ $ep_key ] = $endpoint['title'];
+			}
+
+			$settings[] = array(
+				'class'   => 'llms-select2',
+				'desc'    => '<br>' . __( 'The following LifterLMS Student Dashboard areas will be added to the BuddyPress user profiles', 'lifterlms' ),
+				'default' => array_keys( $display_eps ),
+				'id'      => $this->get_option_name( 'profile_endpoints' ),
+				'options' => $display_eps,
+				'type'    => 'multiselect',
+				'title'   => __( 'User Profile Endpoints', 'lifterlms' ),
+			);
+
+		}
+
+		return $settings;
+
+	}
+
+	/**
+	 * Populate list of endpoints from LifterLMS Dashboard Settings.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function populate_profile_endpoints() {
+
+		$exclude_llms_eps = array( 'dashboard', 'edit-account', 'signout' );
+		$endpoints        = array_diff_key( LLMS_Student_Dashboard::get_tabs(), array_flip( $exclude_llms_eps ) );
+
+		foreach ( $endpoints as $ep_key => &$endpoint ) {
+			unset( $endpoint['nav_item'] );
+			unset( $endpoint['url'] );
+		}
+
+		/**
+		 * Filter profile endpoints.
+		 *
+		 * Modify the LifterLMS dashboard endpoints which can be added to the BuddyPress profile page as custom tabs.
+		 *
+		 * @since [version]
+		 *
+		 * @param array $endpoints Array of endpoint data.
+		 */
+		$this->endpoints = apply_filters( 'llms_buddypress_profile_endpoints', $endpoints );
+
+	}
+
+	/**
+	 * Get a list of custom endpoints to add to BuddyPress profile page.
+	 *
+	 * @since [version]
+	 *
+	 * @param bool $active_only If true, returns only active endpoints.
```

Comment:
Should it be noted that the parameter is optional?

Reply:
Not sure...
We recently "discovered" that the generated html docs  automatically adds "Optional" and the default value. So we thought we could avoid the manual addition, furthermore the manual addition might create duplicates. We wanted to check actually what happens... Do you have the time to make a rapid check with the docs generator?

Reply:
Yes, the generated HTML docs do have "(Optional)" and "Default value:" added, so we don't need to indicate this in the description.

Good example.
https://developer.lifterlms.com/reference/classes/llms_database_query/get/

Redundant example.
https://developer.lifterlms.com/reference/classes/llms_date/db_date/

Reply:
So I'd say we don't add that stuff anymore. At most, if possible, the docs parser should remove `Optional.` and `Default value [...] .` from the description, if we want to keep them in the codebase. But I don't think it's worthwhile. Let's stop using them and progressively remove them from the codebase - if @thomasplevy agrees.

Reply:
I agree, no need to write things that are going to be automatically output.